perun/logic/runner.py

```diff
@@ -473,7 +437,9 @@ def run_postprocessor_on_profile(
 
     p_status, processed_profile = run_postprocessor(postprocessor_unit, profile_job, prof)
     if p_status == PostprocessStatus.OK and prof and not skip_store:
-        store_generated_profile(processed_profile, profile_job)
+        profile.save_profile(
+            profile.finalize_profile_for_job(prof, profile_job, prof.get("header", {}).get("label"))
```

Comment:
This will fail if there is no header.

Reply:
Could you please elaborate this a bit more? The `prof.get("header", {})` is there exactly for situations when a header is not present in a profile, in which case it will use the default value of empty dictionary and as a result, the label parameter will be set to `None`.